osmnx/core.py

```diff
@@ -503,15 +503,23 @@ def gdf_from_places(queries, gdf_name='unnamed', buffer_dist=None):
         later)
     buffer_dist : float
         distance to buffer around the place geometry, in meters
+    which_results : list
+        list (with the same size of queries) of max number of results to return
+        and which to process upon receipt
 
     Returns
     -------
     GeoDataFrame
     """
-    # create an empty GeoDataFrame then append each result as a new row
+    # create an empty GeoDataFrame then append each result as a new row, checking for the presence of which_results
     gdf = gpd.GeoDataFrame()
-    for query in queries:
-        gdf = gdf.append(gdf_from_place(query, buffer_dist=buffer_dist))
+    if which_results != None:
+        assert len(queries) == len(which_results), 'Which_results list length must be the same as queries list length'
```

Comment:
assert len(queries) == len(which_results), 'which_results list length must be the same as queries list length'

Reply:
Done